sway/desktop/output.c

```diff
@@ -401,18 +408,31 @@ bool output_has_opaque_overlay_layer_surface(struct sway_output *output) {
 	return false;
 }
 
-static void send_frame_done_iterator(struct sway_output *output,
+static void send_frame_done_iterator(struct sway_output *output, struct sway_view *view,
 		struct wlr_surface *surface, struct wlr_box *box, float rotation,
 		void *data) {
-	struct timespec *when = data;
-	wlr_surface_send_frame_done(surface, when);
+	int view_max_render_time = 0;
+	if (view != NULL) {
+		view_max_render_time = view->max_render_time;
+	}
+
+	int delay = output->refresh_nsec / 1000000 - output->max_render_time
```

Comment:
This delay doesn't take into account the current time. I think we should compute it relative to the presentation timestamp. Otherwise if a client submits a content update pretty late we'll miss vblank.